fud2/fud-core/src/flang/ir.rs

```diff
@@ -0,0 +1,125 @@
+//! An internal representation of flang.
+
+use camino::Utf8PathBuf;
+use cranelift_entity::{PrimaryMap, entity_impl};
+
+use crate::exec::OpRef;
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct PathRef(u32);
+entity_impl!(PathRef, "path");
+
+#[derive(Debug, PartialEq)]
+pub struct IrAssign {
+    op: OpRef,
+    args: Vec<PathRef>,
+    rets: Vec<PathRef>,
+}
+
+impl IrAssign {
+    pub fn op_ref(&self) -> OpRef {
+        self.op
+    }
+
+    pub fn args(&self) -> &[PathRef] {
+        &self.args
+    }
+
+    pub fn rets(&self) -> &[PathRef] {
+        &self.rets
+    }
+}
+
+impl IrAssign {
+    fn from_parts(op: OpRef, args: &[PathRef], rets: &[PathRef]) -> Self {
+        Self::from_vecs(op, args.to_vec(), rets.to_vec())
+    }
+    fn from_vecs(op: OpRef, args: Vec<PathRef>, rets: Vec<PathRef>) -> Self {
+        Self { op, args, rets }
+    }
+}
+
+/// A flang program.
+#[derive(Default, Debug, PartialEq)]
+pub struct Ir {
```

Comment:
Similarly to the `Step` comment above, this could be called `Plan` to reflect what it fundamentally represents.